plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -427,13 +569,42 @@ public function read( &$order ) {
 				$prop_setter_function_name = "set_{$prop_details['name']}";
 				if ( is_callable( array( $order, $prop_setter_function_name ) ) ) {
 					$order->{$prop_setter_function_name}( $order_data->{$prop_details['name']} );
+				} elseif ( is_callable( array( $this, $prop_setter_function_name ) ) ) {
+					$this->{$prop_setter_function_name}( $order, $order_data->{$prop_details['name']} );
 				}
 			}
 		}
 
 		$order->set_object_read();
 	}
 
+	/**
+	 * Read metadata directly from database.
+	 *
+	 * @param \WC_Order $order Order object.
+	 *
+	 * @return array Metadata array.
+	 */
+	public function read_meta( &$order ) {
+		global $wpdb;
+		$meta_table = $this::get_meta_table_name();
```

Comment:
I believe we should be using `self::get_meta_table_name()` to call class methods, but given this is implemented differently in #33026 I'm not going to worry too much about it.